### PR TITLE
chore(@cypress/react): Rework command logging and snapshotting

### PR DESCRIPTION
**Pull request by dmtrKovalenko**
Changes how we display mount log in command log:

1. Do not show 2 snapshots – only for the final state
2. We do not have problems with snapshot names
3. Little bit better format for console props 
image

### CODE REVIEW

File: npm/react/src/mount.ts [hunk 1]
```diff
@@ -90,28 +79,20 @@ export const mount = (jsx: React.ReactNode, options: MountOptions = {}) => {
...
-
-      logInstance.set('consoleProps', () => logConsoleProps)
-
-      if (el.children.length) {
-        logInstance.set(
-          '$el',
-            (el.children.item(0) as unknown) as JQuery<HTMLElement>,
-        )
-      }
+    if (options.log !== false) {
```
Comment (agg23) [inline npm/react/src/mount.ts:82]: super nit: Compare against `true` unless `options.log` is more than just `boolean | undefined`

**Comment by agg23** [comment]
> Do not show 2 snapshots – only for the final state

Are we sure that this is what we want to do? I thought I saw some discussion asserting that the before state is useful, even if it is a bit weird sometimes.

**Review by agg23 — COMMENTED** [review]
I don't know why this change results in the previous snapshot not showing, so I'll simply leave my very low priority nit

File: npm/react/src/mount.ts (hunk 1 above)
Comment (dmtrKovalenko) [reply npm/react/src/mount.ts:82]: `undefined` means `true` here :D

**Comment by dmtrKovalenko** [comment]
For me, it feels more clear and I don't think it is possible to focus on the second snapshot before the first one

File: npm/react/src/mount.ts (hunk 1 above)
Comment (agg23) [reply npm/react/src/mount.ts:82]: Oh duh. Sorry

**Review by elevatebart — APPROVED** [review]
Do we have the same problem with vue?
Should we fix it the same way?

**Review by lmiller1990 — COMMENTED** [review]
Cool. I agree we should apply this change to `cypress/vue` as well. Might be good to do it in this PR.

**Review by lmiller1990 — APPROVED** [review]

**Pull request merged**